code/datums/components/bumpattack.dm

```diff
@@ -0,0 +1,74 @@
+/datum/component/bumpattack
+    dupe_mode = COMPONENT_DUPE_UNIQUE
+
+    var/valid_slots
+    var/active = FALSE
+
+    var/mob/living/wearer
+
+    var/obj/item/proxy_weapon
+
+
+/datum/component/bumpattack/Initialize(valid_slots, obj/item/proxy_weapon)
+    if(!isitem(parent))
+        return COMPONENT_INCOMPATIBLE
+
+    src.valid_slots = valid_slots
+
+    if(proxy_weapon)
+        src.proxy_weapon = proxy_weapon
+
+
+/datum/component/bumpattack/Destroy(force, silent)
+    return ..()
+ 
+/datum/component/bumpattack/RegisterWithParent()
+    RegisterSignal(parent, COMSIG_ITEM_EQUIPPED, .proc/check_equip)
+    RegisterSignal(parent, COMSIG_ITEM_DROPPED, .proc/check_drop)
+ 
+/datum/component/bumpattack/UnregisterFromParent()
+    UnregisterSignal(parent, list(COMSIG_ITEM_EQUIPPED, COMSIG_ITEM_DROPPED))
+ 
+/datum/component/bumpattack/proc/check_equip(obj/item/source, mob/living/user, slot)
```

Comment:
Document the procs using dmdoc (see my comment above `/datum/component/bumpattack/proc/deactivate()`)